server/pulp/server/managers/consumer/agent.py

```diff
@@ -404,6 +418,46 @@ def _unbindings(bindings):
             agent_bindings.append(agent_binding)
         return agent_bindings
 
+    @staticmethod
+    def delete_queue(url, name, consumer_id):
+        """
+        Delete the agent queue.
+        :param url: The broker URL.
+        :type url: str
+        :param name: The queue name.
+        :type name: str
+        :param consumer_id: The consumer ID.
+        :type consumer_id: str
+        """
+        try:
+            manager = managers.consumer_manager()
+            manager.get_consumer(consumer_id)
+            return  # still registered (abort)
+        except MissingResource:
+            # expected
+            pass
+        agent = PulpAgent()
+        agent.delete_queue(url, name)
+
+
+@task(base=Task)
```

Comment:
This will cause the task to be recorded with a task status. That is what you want right? Can you verify that the task_status records that get created look right? If you want it to occur but not be user facing then you can not use `base=Task`